src/rpc/tungstenite_client/client.rs

```diff
@@ -134,9 +134,25 @@ fn subscribe_to_server(
 	// Subscribe to server
 	socket.send(Message::Text(json_req))?;
 
+	// Read the first message response - must be the subscription id.
+	let msg = read_until_text_message(&mut socket)?;
+	let value: Value = serde_json::from_str(&msg)?;
```

Comment:
As far as I understand, all the following logic is analyzing a string. I would put this logic into its own function and write a test for each flow through. Then you know, that it's doing what you expect.

Reply:
I agree, unit testing does make sense here. But not just here, but also in the ws_client and other functions of this client. Would you agree to do that in a separate issue and PR?

Reply:
I would add tests for new logic from this PR in this PR. For adding tests to existing code, I would create a new issue.

Reply:
Added issue: https://github.com/scs/substrate-api-client/issues/663, commit adding unit tests for changed code is incoming.